packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -120,6 +155,140 @@ function readBlobContent(content: ISummaryBlob["content"]): unknown {
 	return JSON.parse(json);
 }
 
+export class TestFluidObjectFactory implements IFluidDataStoreFactory {
+	public static readonly type = "@fluid-example/test-example";
+	public readonly type = TestFluidObjectFactory.type;
+
+	public get IFluidDataStoreFactory() {
+		return this;
+	}
+
+	public async instantiateDataStore(
+		context: IFluidDataStoreContext,
+		existing: boolean,
+	): Promise<FluidDataStoreRuntime> {
+		const runtimeClass = mixinRequestHandler(
+			async (request: IRequest, rt: FluidDataStoreRuntime) => {
+				const maybeRouter: FluidObject<IProvideFluidRouter> | undefined =
+					await rt.entryPoint?.get();
+				assert(
+					maybeRouter?.IFluidRouter !== undefined,
+					"entryPoint should have been initialized by now",
+				);
+				return maybeRouter.IFluidRouter.request(request);
+			},
+		);
+
+		return new runtimeClass(context, new Map(), existing, async () => {
+			throw new Error("EntryPoint not implemented");
+		});
+	}
+}
+
+export class TestDataObjectContext implements IFluidDataStoreContext {
```

Comment:
I am not sure I understand why a mock data store context is created. How is this helping the validation you are trying to add?

Reply:
I actually do need this real object context to let runtime factory to create a requestable container.

Reply:
This is an end-to-end test so you can use a real data object and data object factory. Isn't the test validating that the finishInitizalition function on DataObjectFactory is not called?

Reply:
Seems like previous test case use summarizer to load existing container, so I don't have chance to define my own testing entrypoint function.

Reply:
Sure, but you don't need to create your own data object factory and data store context to do that. This is not testing what you think its testing. The purpose of the test seems to be that PureDataObject::finishInitialization is not called unless getEntryPoint / request is called on it. What your test is doing is its validating the finishInitialization equivalent of this mock test factory is not called.

Reply:
You will have to create a real data object / pure data object, call instantiateDataStore on it and validate that finishInitialization is not called on it. Yes, you cannot create your own entry points but that's not the goal.
What you can do is create a class that extends `DataObject` and add an assert / throw error in its `hasInitialized` function. Then call `instantiateDataStore` on its data object factory and validate that the assert in `hasInitialized` is not hit.